bin/cli.js

```diff
@@ -1,17 +1,172 @@
+  });
+  return config;
+};
+
+const setKarmaLogLevel = (config, logLevel = 'disable') => {
+  if (!config.webpackMiddleware) config.webpackMiddleware = {};
+  config.logLevel = karma.constants[karmaLogLevels[logLevel][0]];
+  config.webpackMiddleware.stats = karmaLogLevels[logLevel][1];
+  return config;
+};
+
+const createFactory = () => {
+  const factory = (files) => {
+    files.unshift({ pattern: path.resolve(__dirname, '../karma/adapter.js') });
+  };
+  factory.$inject = ['config.files'];
+  return factory;
+};
 
-walk('../test', (err, files) => {
-  metasync.for(files).each((file, next) => {
-    require(file.filePath);
-    next();
-  }).fetch(() => {
-    console.log('Tests done');
+const getBrowserConfig = (conf) => {
+  const config = {
+    files: [],
+    preprocessors: {},
+    plugins: [
+      { 'framework:metatests': ['factory', createFactory()] },
+      'karma-webpack'
+    ],
+    concurrency: 1,
+    autoWatch: false,
+    singleRun: true,
+    frameworks: ['metatests'],
+    basePath: process.env.PWD,
+    webpack: { node: {} },
+    webpackMiddleware: {},
+    port: conf.browser.port
+  };
+  setKarmaBrowsers(config, ...conf.browser.browsers);
+  setKarmaLogLevel(config, conf.browser.logLevel);
+  conf.files.forEach(file => {
+    config.files.push({ pattern: file, type: 'module' });
+    config.preprocessors[file] = [ 'webpack' ];
   });
-});
+  [ 'fs', 'child_process' ].forEach(lib => {
+    config.webpack.node[lib] = 'empty';
+  });
+  return config;
+};
+
+const getConfig = () => {
+  const version = parseFile(path.resolve(__dirname, '../package.json')).version;
+  program.version(version).usage('[options] -- <file ...>');
+  clOptions.forEach(option => program.option(...option));
+  program.parse(process.argv);
+
+  const config = program.config ? parseFile(program.config) : {};
+
+  config.enviroment = program.enviroment;
+  config.files = config.files || [];
+  config.files.push(...program.args);
+  config.files = loadFiles(config.files);
+  config.packages = config.packages || [];
+  config.packages.push(...program.requiredPackages);
+  config.packages = rmDuplicates(config.packages);
+
+  if (config.enviroment === 'browser') {
+    config.browser = config.browser || {};
+    config.browser.browsers = config.browser.browsers || [];
+    config.browser.browsers.push(...program.browsers);
+    config.browser.logLevel = config.browser.logLevel || program.browserLog;
+    config.browser.port = +program.browserPort || config.browser.port || 9876;
+    config.browser = getBrowserConfig(config);
+  }
+  return config;
+};
+
+const runNode = (config, cb) => {
+  global.api = { metatests: require('..') };
+  config.packages.map(name => global.api[name] = require(name));
+  config.files.map(name => require(path.resolve('./' + name)));
+  global.api.metatests.runner.instance.on('finish', (failed) => {
+    console.log('Tests finished. Waiting for unfinished tests after end...');
+    setTimeout(() => cb(+failed), 5000);
```

Comment:
This is still not addressed, you should not rely on the type coercion from bool to int and directly specify the result.
`hasFailures ? 1 : 0`